src/main/java/org/elasticsearch/cluster/metadata/MappingMetaData.java

```diff
@@ -354,7 +354,7 @@ private void initMappers(Map<String, Object> withoutType) {
                     path = fieldNode.toString();
                 } else if (fieldName.equals("format")) {
                     format = fieldNode.toString();
-                } else if (fieldName.equals("default")) {
+                } else if (fieldName.equals("default") && fieldNode != null) {
```

Comment:
original reporter here who discovered this issue after using _timestamp with default=null. 

For "default" field in particular, *null* actually means something (which is why i used it) - to ignore the normal _timestamp default of *now* if _timestamp is not provided, thereby making _timestamp (or the field providing it via *path* property) a **required field**.  seems like "default" property needs separate handling from other fields that might inadvertently have been passed a *null* - as in this case passing a null has a meaning (that being: clear any default behavior that the _timestamp should have).

From [_timestamp doc](http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/mapping-timestamp-field.html), section on Default.  
>By default, the default value is *now* which means the date the document was processed by the indexing chain. 
>
> You can disable that default value by setting default to *null*. It means that timestamp is mandatory: